### PR TITLE
RUMM-2320 JSON schemas for IncrementalData types

### PR DESCRIPTION
**Pull request by mariusc83**
This PR adds the remaining JSON schemas for the `IncrementalData` types which will go in the `IncrementalSnapshotRecord#data`.

### CODE REVIEW

**Review by mquentin — APPROVED** [review]
LGTM

did you changes the blocks you extracted in their own json files like `schemas/session-replay/mobile/text-style-schema.json` ? (to see if I need to re-generate the shape and text wireframes types)

**Comment by mariusc83** [comment]
> LGTM
> 
> did you changes the blocks you extracted in their own json files like `schemas/session-replay/mobile/text-style-schema.json` ? (to see if I need to re-generate the shape and text wireframes types)

Nope, I just extracted them, nothing was changed.

File: schemas/session-replay/mobile/mutation-data-schema.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    "description": "The source of this type of incremental data.",
+                    "readOnly": true
+                },
+                "adds": {
+                    "type": "array",
+                    "readOnly": true,
+                    "items": {
+                        "type": "object",
+                        "required": [
+                            "previousId",
```
Comment (ncreated) [inline schemas/session-replay/mobile/mutation-data-schema.json:22]: We need `previousId` to accept `null` value, hence it cannot be `required`. This is to enable placing the new `wireframe` at the beginning of the array.

Ref. in RFC: [Adding new wireframes](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/2471460887/RFC+-+Data+Format+for+Mobile+Session+Replay)
> wireframe in draw order (or null if the new wireframe should be rendered at the very bottom).

File: schemas/session-replay/mobile/mutation-data-schema.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                            "wireframe"
+                        ],
+                        "properties": {
+                            "previousId": {
+                                "type": "integer",
+                                "description": "The previous wireframe id next/after which this new wireframe should be drawn/attached to.",
+                                "readOnly": true
+                            },
```
Comment (ncreated) [inline schemas/session-replay/mobile/mutation-data-schema.json]: I'm not sure this is correct. We need the `previousId` to support `null` value in order to place new `wireframe` at the beginning of the array.

Ref. in RFC: [Adding new wireframes](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/2471460887/RFC+-+Data+Format+for+Mobile+Session+Replay)
> Besides the wireframe to add, it holds the ID of the previous wireframe in draw order (or null if the new wireframe should be rendered at the very bottom).

So, the allowed values for `previousId` should be: `Int | null`. Doesn't it mean we need to enhance the `type` definition for `previousId`? I'm not sure how to model this in [JSON schema](https://json-schema.org/understanding-json-schema/reference/object.html#id8). 

@mariusc83  @BenoitZugmeyer any thoughts?

Comment (BenoitZugmeyer) [reply schemas/session-replay/mobile/mutation-data-schema.json]: Good catch. Indeed, if you want to put the wireframe at the very beginning of the list, `previousId` should not be defined. IMO an optional property would be fine, no need to have `Int | null` explicitely.

Comment (ncreated) [reply schemas/session-replay/mobile/mutation-data-schema.json]: 👍 Let's make it not `required` then. In fact, in Swift-generated code both `Int | null | none` are expressed with optional, so no preference on our side.

Comment (mariusc83) [reply schemas/session-replay/mobile/mutation-data-schema.json]: Yup, good catch, we will make it not required.

**Review by ncreated — APPROVED** [review]
Looks good 👌!

**Pull request merged**